src/Types.hs

```diff
@@ -70,15 +68,15 @@ runStrategy name strategyGroup act = forkTask $ do
         let (bodies, ()) = resultValue result
          in traverse_ (output . toProjectClosure strategyGroup name) bodies -- TODO: warnings
 
-type TaskC m = ExecIOC (ReadFSIOC (DiagnosticsC m))
```

Comment:
Rather than using `ExecIOC` and `ReadFSIOC` as part of running the task, we send the constraints upward (see the below additions to `HasDiscover`)